src/lib/storage/table.hpp

```diff
@@ -91,7 +91,7 @@ class Table : private Noncopyable {
 
  protected:
   const uint32_t _chunk_size;
-  std::vector<Chunk> _chunks;
+  std::vector<std::shared_ptr<Chunk>> _chunks;
```

Comment:
Actually this can still be a `std::vector<Chunk>` without the `std::shared_ptr` as we can replace vector elements in-place (by moving). Our intuition (wrongly) suggested this wasn't possible.